instrumentation/jdbc/library/src/test/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.java

```diff
@@ -0,0 +1,302 @@
+            return false;
+          }
+        };
+    OpenTelemetryDriver.addDriverCandidate(localDriver3);
+
+    Driver winner3 = OpenTelemetryDriver.findDriver("jdbc:test:");
+    OpenTelemetryDriver.removeDriverCandidate(localDriver3);
+
+    assertThat(winner3).isNotNull();
+    assertThat(winner3).isNotEqualTo(localDriver3);
+  }
+
+  @DisplayName("Verify connection with accepted url")
+  @Test
+  void verifyConnectionWithAcceptedUrl() throws SQLException {
+
+    registerTestDriver();
+
+    Connection connection2 = OpenTelemetryDriver.INSTANCE.connect("jdbc:otel:test:", null);
+
+    assertThat(connection2).isNotNull();
+    assertThat(connection2).isExactlyInstanceOf(OpenTelemetryConnection.class);
+  }
+
+  @DisplayName("Verify get property info with test driver url")
+  @Test
+  void verifyGetPropertyInfoWithTestDriverUrl() throws SQLException {
+
+    registerTestDriver();
+    String testUrl = "jdbc:otel:test:";
+    DriverPropertyInfo[] propertyInfos =
+        OpenTelemetryDriver.INSTANCE.getPropertyInfo(testUrl, null);
+    assertThat(propertyInfos).hasSize(1);
+    assertThat(propertyInfos[0].name).isEqualTo("test");
+    assertThat(propertyInfos[0].value).isEqualTo("test");
+  }
+
+  private static void deregisterTestDriver() {
+    if (hasTestDriver()) {
+      try {
+        DriverManager.deregisterDriver(new TestDriver());
+      } catch (SQLException e) {
+        throw new IllegalStateException(e);
+      }
+    }
+  }
+
+  private static void registerTestDriver() {
+    if (!hasTestDriver()) {
+      try {
+        DriverManager.registerDriver(new TestDriver());
+      } catch (SQLException e) {
+        throw new IllegalStateException(e);
+      }
+    }
+  }
+
+  private static boolean hasTestDriver() {
+    Enumeration<Driver> drivers = DriverManager.getDrivers();
+    while (drivers.hasMoreElements()) {
+      Driver driver = drivers.nextElement();
+      if (driver instanceof TestDriver) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @DisplayName("verify get property info with null url")
+  @Test
+  void verifyGetPropertyInfoWithNullUrl() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> OpenTelemetryDriver.INSTANCE.getPropertyInfo(null, null),
+        "url is required");
+  }
+
+  @DisplayName("verify get property info with empty url")
+  @Test
+  void verifyGetPropertyInfoWithEmptyUrl() {
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> OpenTelemetryDriver.INSTANCE.getPropertyInfo(" ", null),
+        "url is required");
+  }
+
+  @DisplayName("verify get property info with unknown driver url")
+  @Test
+  void verifyGetPropertyInfoWithUnknownDriverUrl() {
+    String unknownUrl = "jdbc:unknown";
+    assertThrows(
+        IllegalStateException.class,
+        () -> OpenTelemetryDriver.INSTANCE.getPropertyInfo(unknownUrl, null),
+        "Unable to find a driver that accepts url:" + unknownUrl);
```

Comment:
which makes me think that the message is not being asserted against?

```suggestion
        "Unable to find a driver that accepts url: " + unknownUrl);
```